### PR TITLE
feat: applicant respond to respondent's defence (CMC-424)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-424

### Change description ###

New respond to defence event for solicitor idam role.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by AlisonRevittHM — DISMISSED** [review]
data review

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.unspec.callback.CaseEvent.CLAIMANT_RESPONSE;
+
+@Service
+@RequiredArgsConstructor
+public class RespondToDefenceCallbackHandler extends CallbackHandler {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: Will this be only response to full defence type of the defendant response? how about part admit, counter claim, full admit etc?

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    JacksonAutoConfiguration.class
+})
+class RespondToDefenceCallbackHandlerTest extends BaseCallbackHandlerTest {
+
+    @Autowired
+    private RespondToDefenceCallbackHandler handler;
+
+    @Nested
+    class SubmittedCallback {
+        public static final String APPLICANT_1_PROCEEDING = "applicant1ProceedWithClaim";
```
Comment (sabahirfan) [inline src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandlerTest.java:33]: can we use this from `RespondToDefenceCallbackHandler` itself?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java (hunk 1 above)
Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: counts for everything, I think this wording makes sense to. The defence is any of those options above. The claimant then responds to that defence

File: src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandlerTest.java (hunk 2 above)
Comment (swalker125) [reply src/test/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandlerTest.java:33]: no, I makes the test more meaningful if this is also defined here imo

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java (hunk 1 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: Full admission won't be the defence.

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: doesn't mention any differences in the JIRA ticket so for now it is as is.

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: and sure full admission is still a type of defence 😅 but yeah, I have already added a comment to the JIRA ticket about. how this event should handle different defence types as my interpretation is it handles all, but maybe this is not correct

**Review by sabahirfan — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java (hunk 1 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/RespondToDefenceCallbackHandler.java:23]: The way I read it, it's `Defendant Response` and not the defence 🤔 🤣 
 which has types of `Full Defence`, 'Full Admission`, `Part Admission` and `Counter Claim`.

**Review by douglasrice — DISMISSED** [review]

**Review by sabahirfan — DISMISSED** [review]

**Review by douglasrice — APPROVED** [review]

**Review by sabahirfan — APPROVED** [review]

**Pull request merged**